packages/template-retail-react-app/app/pages/registration/index.jsx

```diff
@@ -7,36 +7,55 @@
 
 import React, {useEffect} from 'react'
 import PropTypes from 'prop-types'
+import {useIntl} from 'react-intl'
 import {Box, Container} from '@chakra-ui/react'
-import useCustomer from '../../commerce-api/hooks/useCustomer'
-import Seo from '../../components/seo'
+import {
+    ShopperLoginHelpers,
+    useShopperLoginHelper,
+    useCustomerType
+} from 'commerce-sdk-react-preview'
 import {useForm} from 'react-hook-form'
+import {useLocation} from 'react-router-dom'
+import Seo from '../../components/seo'
 import RegisterForm from '../../components/register'
 import useNavigation from '../../hooks/use-navigation'
 import useEinstein from '../../commerce-api/hooks/useEinstein'
-import {useLocation} from 'react-router-dom'
+import {API_ERROR_MESSAGE} from '../../constants'
 
 const Registration = () => {
+    const {formatMessage} = useIntl()
     const navigate = useNavigation()
-    const customer = useCustomer()
+    const customerType = useCustomerType()
     const form = useForm()
     const einstein = useEinstein()
     const {pathname} = useLocation()
+    const register = useShopperLoginHelper(ShopperLoginHelpers.Register)
 
-    const submitForm = async (data) => {
-        try {
-            await customer.registerCustomer(data)
-        } catch (error) {
-            form.setError('global', {type: 'manual', message: error.message})
+    const submitForm = (data) => {
+        const body = {
+            customer: {
+                firstName: data.firstName,
+                lastName: data.lastName,
+                email: data.email,
+                login: data.email
+            },
+            password: data.password
         }
+
+        return register.mutateAsync(body, {
+            onSuccess: () => navigate('/account'),
+            onError: () => {
+                form.setError('global', {type: 'manual', message: formatMessage(API_ERROR_MESSAGE)})
+            }
+        })
     }
 
     // If customer is registered push to account page
     useEffect(() => {
-        if (customer.authType != null && customer.isRegistered) {
+        if (customerType === 'registered') {
             navigate('/account')
         }
-    }, [customer])
+    }, [])
```

Comment:
```suggestion
    }, [customerType])
```
Do we need this here? (React is magic, I don't understand everything. 🤷 )

Reply:
We don't need this, `navigate` is async in a sense that the components are not unmounted immediately. I ran into a situation where after navigation, this component still gets update from hooks and causing multiple navigate calls.

The solution is to move `navigate` call to the api onsuccess handler, and only have this useEffect run on mount ONLY ONCE, which we can use an empty dependency array.

Reply:
Are we sure we aren't user this alternative solution to work around a bug elsewhere?
If the dependency list was `[isRegistered]` I would not expect the effect to be called more than once after the state went to false to true. 
I feel like navigating as a reaction to the state changing is better react practice, it will also dry you code since you don't have navigate called in 2 places. But I can appreciate you wanting to have symmetrical code with onSuccess and onFailure.

Reply:
I tried it again, and @bendvc you are right, `[isRegistered]` works like a charm... I must did something wrong when i implemented this halfway. 

Fixed!